src/constructors/createGlobalStyle.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+import React from 'react'
+import PropTypes from 'prop-types'
+import hashStr from '../vendor/glamor/hash'
+import StyleSheet, { CONTEXT_KEY } from '../models/StyleSheet'
+import ServerStyleSheet from '../models/ServerStyleSheet'
+import type { Interpolation, Stringifier } from '../types'
+import {
+  CHANNEL,
+  CHANNEL_NEXT,
+  CONTEXT_CHANNEL_SHAPE,
+} from '../models/ThemeProvider'
+
+export default (stringifyRules: Stringifier, css: Function) => {
+  const createGlobalStyle = (
+    strings: Array<string>,
+    ...interpolations: Array<Interpolation>
+  ) =>
+    class extends React.Component {
+      static contextTypes = {
+        [CHANNEL]: PropTypes.func,
+        [CHANNEL_NEXT]: CONTEXT_CHANNEL_SHAPE,
+        [CONTEXT_KEY]: PropTypes.oneOfType([
+          PropTypes.instanceOf(StyleSheet),
+          PropTypes.instanceOf(ServerStyleSheet),
+        ]),
+      }
+      state = { theme: {} }
+      componentId = ''
+      unsubscribeId = -1
+
+      componentWillMount() {
+        const context = this.context[CHANNEL_NEXT]
```

Comment:
It doesn't quite look like this is matching our usual theming/subscription logic. We have some code for it here (it's easier to read in `withTheme`): https://github.com/styled-components/styled-components/blob/master/src/hoc/withTheme.js#L40-L60

It'd be cool if we could make a common base class for this stuff. Let's take a look at that in a separate PR though